### PR TITLE
Dev/fe

### PR DESCRIPTION
**Pull request by ngocdu171**
install dependencies

### CODE REVIEW

**Review by thinhnguyen5 — APPROVED** [review]

**Comment by toilavic** [comment]
NOTE: No merge until everything on this brand works perfectly

**Review by toilavic — COMMENTED** [review]
LGTM!

**Review by toilavic — COMMENTED** [review]
test

File: frontend/src/data/stores/interfaceStore.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    rate: [],
+    price: number,
+    openTime: string,
+    contactNumber: string,
+    description: string,
+    discountRate: number,
+    instagram: string,
+    createdAt: string,
+    updateAt: string,
+    lat: number,
```
Comment (toilavic) [inline frontend/src/data/stores/interfaceStore.ts:14]: why do you duplicated those interfaces? What is the purpose of IStores?

File: frontend/src/data/stores/stores.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import interfaceStore from "./interfaceStore";
```
Comment (toilavic) [inline frontend/src/data/stores/stores.ts:1]: You can still using the mock data for the open-map. But make sure to delete it when we have the real data from backend side. 
Please check our the Backend tickets in Jira

File: frontend/src/data/users/interfaceUser.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default interface interfaceUser {
```
Comment (toilavic) [inline frontend/src/data/users/interfaceUser.ts:1]: Remember to checkout the backend ticket in Jira. The constructor will be changed with the new type and permission

File: frontend/src/data/users/users.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import interfaceUser from "./interfaceUser";
```
Comment (toilavic) [inline frontend/src/data/users/users.ts:1]: Remove the mock data

File: frontend/src/pages/login/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+const Login = () => {
+  const {Login, auth} = useContext(StoresContext)
+  const [email, setEmail] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+
+  const handleSubmit = (event: ChangeEvent<HTMLInputElement>) => {
+    //console.log("email: ", email, "password: ", password);
+    //Login(email,password);
+    //console.log(email, password);
+    Login(email, password);
```
Comment (toilavic) [inline frontend/src/pages/login/Login.tsx:24]: when apply the new API, please remember to use async await to wait for the responce.

File: frontend/src/pages/map/MapPage.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+          <Grid item xs={5}>
+            <Showmap stores = {stores} test = {test} />
+          </Grid>
+        </Grid>
+      </Box>
+      <Box>
+        <Grid container>
+          <Grid item xs={12}>
+            <Grid>
+            {test.map((tam) => {
```
Comment (toilavic) [inline frontend/src/pages/map/MapPage.tsx:93]: Meaningless name variables

File: frontend/src/pages/register/Register.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+import { Link } from "react-router-dom";
+
+const theme = createTheme();
+
+const Register: FC = () => {
+
+    const [email, setEmail] = useState<string>();
+    const [password, setPassword] = useState<string>();
+
+    const handleRegister = (event:ChangeEvent<HTMLInputElement>) => {
```
Comment (toilavic) [inline frontend/src/pages/register/Register.tsx:21]: remember to add register API to the callback. Use async await for the backend

**Review by toilavic — CHANGES_REQUESTED** [review]
Check our those comments and fix it. 
After fix one comment, push to the dev/FE, and comeback to the corresponding comment then mark them as resolved. I will check it again in the new commit

File: frontend/src/components/showmap/Showmap.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+const Showmap: React.FC<Props> = ({ stores, test }) => {
+    const MAPBOX_TOKEN = "pk.eyJ1IjoiYnJvdGhlcmQiLCJhIjoiY2t2NnYyeTN6MWc2ejJubzA3dmE1ajdsYSJ9.D2EEYvFB43G4_0JQYJn63w";
+
+    const [viewport, setViewport] = useState({
+        latitude: 65.0118734,
+        longitude: 25.4716809,
+        zoom: 12,
+    });
+
+    const [selectedRestaurant, setSelectedRestaurant]:any = useState(null);
```
[the remark this answers is not in the document]
Comment (ngocdu171) [reply frontend/src/components/showmap/Showmap.tsx:34]: done

File: frontend/src/components/showmap/Showmap.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+                            </CardContent>
+                        </Card>
+                    </Popup>
+                ) : null}
+            </ReactMapGL> */}
+
+            <ReactMapGL
+            {...viewport}
+            mapboxApiAccessToken={MAPBOX_TOKEN}
+            height = {650}
```
[the remark this answers is not in the document]
Comment (ngocdu171) [reply frontend/src/components/showmap/Showmap.tsx]: define mapbox token in constant.tsx

**Comment by toilavic** [comment]
We have a huge code debt in this branch, and also a lot of meaningless commits and separated PRs.
Need an update for this branch

**Comment by toilavic** [comment]
_Always a good idea to try to step into someone else's shoes_. "Do I understand what this variable means?" "Do I understand what this code is doing?"  "Do I understand what kind of commits are doing?"
**Making the those things descriptive never hurts anyone!!!**

**Comment by toilavic** [comment]
Close, now use hotfix to be primary branch

**Pull request closed without merging**